dds/DCPS/transport/framework/TransportClient.cpp

```diff
@@ -496,7 +496,8 @@ TransportClient::PendingAssoc::initiate_connect(TransportClient* tc,
           data_.remote_reliable_, data_.remote_durable_};
 
         TransportImpl::AcceptConnectResult res;
-
+        GuidConverter tmp_local(tc->repo_id_);
+        GuidConverter tmp_remote(this->data_.remote_id_);
```

Comment:
tc is a simple TransportClient*, and during the connect_datalink call (inside tc->initiate_connect_i) the guard is released at which time the TransportClient could go away and the connect fails.  This path is for failure to connect/initiate_connect_i path and thus should not access the tc pointer after this call if AcceptConnectResult returned was false.  This allows the stack to unwind without accessing anything inside tc.